### ISSUE TITLE
[2.6.4] No notifications shown on new message receive

### ISSUE THREAD

**Issue by nRaecheR** [WhisperSystems/TextSecure#2703]
After updating to version 2.6.4 from 2.5.x via Google Play Store no notifications about new received messages were shown when application runs in the background. The notifications and the messages are only received when running the the foregound, which leads to missed messages. 

This is new with 2.6.4, I never had the problem with any previous version.

Samsung GT-i9505 (jfltexx)
Android 4.4.4 (KOT49H.I9505XXUGNK4)
OS Host: SWDD6301
TextSecure 2.6.4

**Comment by rhodey** [comment]
unable to reproduce on my Nexus 5 running stock Android Lollipop and TS 2.6.4, could be KitKat specific? anyone else with a KitKat device able to reproduce this? @nRaecheR does this happen with both SMS and push messages?

steps I took to test:

1) unlock TextSecure
2) open recent activities view
3) force close (swipe) TextSecure
4) have @mcginty send me test push and SMS messages
5) phone vibrates and LED blinks for each message, also notifications exist for each.

**Comment by karlhedderich** [comment]
I am having this problem too.  I am on moto X 2nd gen android 4.4.4

**Comment by nRaecheR** [comment]
Mh, it seems that it works most of the time but sometimes the notifications are missing. I think it happened when switching from Wifi internet to mobile/VPN access to the same network but I'm not sure. 

Tested only with push messages, I don't use SMS anymore.

**Comment by crash2kx** [comment]
I've been having the same issue. Sent a secure message to my friend last night a 1.31am, he replied at 1.49am (from his iPhone) and I only received the message once I opened TextSecure this morning.

This issue has been happening several times over the past few days with other contacts as well. The last few times I missed notifications I was in a club (so my guess is spotty reception) and I would only see the message once I forgrounded the app the next day. This time, however, it happened at home.

I really hope this issue will be fixed very soon. Not being able to receive notifications at times is a death sentence for this app as it makes it impossible for me to convince my friends to switch over.

Thanks!

https://gist.github.com/anonymous/e370c390350063a9169a

**Comment by agrajaghh** [comment]
I don't think its a problem with notifications, looks more like a GCM issue...

**Comment by crash2kx** [comment]
Yes, it doesn't look like the messages are received but the notification is skipped but rather that the messages are never received at all until the app is opened manually. Interestingly, last time I had this issue on Sunday once I opened the app I received several messages from the same contact at once that he had sent the night before. So it's not message specific.

**Comment by agrajaghh** [comment]
Is it never working or just sometimes not?

**Comment by crash2kx** [comment]
Just sometimes not. Most of the time it works perfectly.

**Comment by crash2kx** [comment]
It just happened again. I didn't receive two messages until I opened the application and once I did that two messages from 48 minutes ago came in.

https://gist.github.com/a4d2be5ac699d378f20c

**Comment by moxie0** [comment]
If this is happening, it means GCM push messages aren't making it to your device.  The most common reasons for this are:

1. You've disabled "background data" for TextSecure or for Google Play Services in Android's system settings.
2. You've used a tool like Titanium Backup with TextSecure.
3. You're running a custom ROM with a bad gapps install.

**Comment by crash2kx** [comment]
Thanks for the reply!
This is weird, because I am running pure stock Android without any modifications (except root), no Greenify or other mods that could deactivate background data. All other apps (Twitter, WhatsApp, Gmail) push things just fine. Plus, TextSecure works fine most of the time anyway.

**Comment by zmanian** [comment]
So GCM is not reliable message transport and is generally becoming less
reliable.

The fix that OWS put in place is check via websocket for any messages that
are on disk when a GCM message is received or when you open the app.

You will experience this as having an unread message that appears when you
open the app that no notification appears for.

I see this behavior in twitter, fb messenger and other messaging apps
because it seems like most dev teams are mitigating GCM reliability issues
in the same way.

**Comment by moxie0** [comment]
@crash2kx Does this happen more often on one network rather than another? Do you have a VPN or anything like that configured?

**Comment by crash2kx** [comment]
I'm on my home WiFi right now, no VPN configured. But it also happened to me while I was outside on my network's 3G this weekend. It usually works, both on WiFi and 3G. I remember receiving WhatsApp messages just fine during the same time frame while the TextSecure messages were missing.

**Comment by Hnasar** [comment]
Just want to chime in that this is occasionally (i.e. on and off for a few days now) happeneing to me and my friend as well. I'm on CM 12 running the recent beta, she is on xperia stock 4.4 running the most recent non-beta version.

I haven't tracked whether it happened to me when on wifi or cell network. I'll keep an eye out.

**Comment by moxie0** [comment]
It's also entirely possible that this is a client bug.  What we'd need to figure it out is a debug log that covers the *missed* notification.  Steps would be:

1. Before you open the app, you somehow notice that you're missing notifications.
2. Without yet opening the app, have someone send you a message.
3. Wait and see if a notification appears.
4. If you're not notified, open the app (the message should then appear) and *immediately* capture a debug log.

The key is that all this happens before you open the app.

**Comment by crash2kx** [comment]
My second debug log I posted here https://gist.github.com/a4d2be5ac699d378f20c was captured right after I opened the app and received the missed messages (which had been sent 48 minutes before). I hadn't even opened the message yet. Perhaps something can be found in there.  It is difficult to replicate the problem as receiving messages works most of the time, but I will try and do it in case I know the bug is happening. Once a message does not get delivered all following messages from that contact don't get delivered either until the app is opened. I don't know if a message from another contact after the bug has occured will trigger a notification, however.

**Comment by moxie0** [comment]
@crash2kx Yep, unfortunately in that scenario we need the debug log from 48 minutes earlier.  I realize that it's difficult to reproduce, since it requires realizing that you're missing notifications *before* you open the app.

**Comment by crash2kx** [comment]
As luck would have it, I asked my friend to send me a test message and it arrived without a hitch. I'll stay on it. Kinda weird to be hoping for a message not to get through to be able to capture a debug log :D

**Comment by TheStash** [comment]
I had somehow a similar issue but was not sure if it was associated with the VPN or not, the VPN is Bitmask's demo so it should not block GCM but is a bit slow, in my case with two Nexus 5 running Lolipop on the same network one received the group messages and the other didn't unless either fully rebooting the device or manually opening the app... i was actually waiting not to open an issue until i find out if it is associated with the VPN or not.

**Comment by AndrewTheLott** [comment]
Just to confirm, I've been getting the same on my OnePlus One with the default/current Cyanogen 11.0 (4.4.4). There's no VPN running when it happens, but I haven't noticed whether it happens more on Wi-Fi than off.

**Comment by jeremymasters** [comment]
Just opened the app to messages my brother sent me awhile ago. It chirped at me like I had just received them but I know it was awhile ago. Here is a log.https://gist.github.com/a4d48273e45a664d09b9

**Comment by moxie0** [comment]
@jeremymasters Thanks, but we need the log from "awhile ago." If you know someone has sent you a message, but you haven't received it, ask them to send another one before opening the app and taking the log.

**Comment by jeremymasters** [comment]
@moxie0 How far do these logs stretch back?  From looking at mine it's grabbing an mms which was about 6 hours ago through now which would include the time of the messages that were not "received" until I opened the app.

**Comment by jeremymasters** [comment]
If the app was open the whole time, does that change things?

**Comment by jeremymasters** [comment]
Sorry to keep bombarding here, but last night I had the "sender" version of the issue last night.  I sent about 4 messages to someone and the backgrounds all turned bright blue.  I did not get delivery receipts though for about 10 minutes.  I was connected to wifi at the time.  I decided to shut off wifi and switch to 4G.  Right after I turned off my wifi and it registered as off, all 4 of my messages got the delivery receipt and I received another message from my friend, all within 1 second.  After this happened, I eventually turned wifi back on and communication went as normal.  I don't know exactly if/how this is linked but thought maybe it would at least help troubleshoot.

**Comment by crash2kx** [comment]
Okay, what happened now was that I had the app foregrounded for most of the time, checking for new messages from my friend whom I was waiting for to reply, but I still didn't receive any new message. It was only until I sent my friend another message asking "where are you" that I suddenly received two messages, one from 29 minutes ago saying "I'll be there in 25 minutes" and a fresh one just now saying "I'm here". I don't know if the debug actually caught it this time, but here it is anyway.
https://gist.github.com/anonymous/d80b5d7d7a1a131ca05c

**Comment by merkste** [comment]
Unfortunately I can confirm the issue since the update to 2.7.1 but cannot provide any new context info.

**Comment by moxie0** [comment]
@merkste Yeah, 2.7.1 doesn't include any changes in this area.  I just pushed 2.8.0 to the beta channel, though, and I'll be curious to see if that has any effect.

**Comment by moxie0** [comment]
I just pushed some server-side changes that might help to production 10min ago, please let me know if anyone continues to see this on 2.7.1 or 2.8.0. Thanks.

**Comment by jeremymasters** [comment]
Will do.  Sorry if I've been annoying about this.  I know you're neck-deep in work.

**Comment by crash2kx** [comment]
I got a message 3 minutes ago and it notified me as normal. When I picked up my phone to respond I opened the app and immediately got a second message. Interestingly enough it was also dated 3 minutes ago, so my friend had apparently sent 2 messages in a row, but only the first one had come through normally.
Hope the debug log goes back enough to catch the bug occuring in the 3 minute time frame this time. Would love for this bug to get fixed.

https://gist.github.com/anonymous/d06df7fbcaa168ea9482

**Comment by moxie0** [comment]
@crash2kx If you're comfortable doing it, would you mind emailing me your phone number and the timestamp on that delayed message? moxie.github@moxie.org

**Comment by gordon-morehouse** [comment]
I'm still seeing these issues on 2.7.1. I tried to dump a debug log this morning but I got nothing but the usual header (android and TS version, etc). They are still happening regardless of my data connectivity (typically NATed wifi on a couple different Comcast connections, and 3G).

Sometimes messages will stop going through (with me as the sender - I stop getting delivery confirms) in the middle of an active conversation. It is tough to coordinate to get debug logs there, and it does happen less often, but often enough (in my experience) that some users are likely to consider the app unreliable.

Before secure SMS transport was removed, falling back to secure SMS appeared to ram a message through if the data transport was not working. I'm not sure that works with insecure SMS.

**Comment by moxie0** [comment]
> Before secure SMS transport was removed, falling back to secure SMS appeared to ram a message through if the data transport was not working.

FYI, it did no such thing.

**Comment by linusbergl** [comment]
I am still experiencing this on the 2.8 beta. From textsecure and signal. Will try to send a debug log according to moxie's description.

**Comment by Sinlicious** [comment]
Just signed up to chime on this - from what I'm seeing it could be a mixture of 'delayed' messages (#970) or simply not being notified as this issue would suggest. What is happening is; I'm having to open up TextSecure to 'jump start' any notifications/messages received - the weird thing is, they are not sitting there 'bolded' as if I didn't get the notification, but more-so actually appear when I'm killing the application and opening TextSecure again (which makes me think they are being delayed/late).

I'm using 2.7.1 on a Android 5.0.2 - Hangouts and jazz has always been fine (which obviously utilizes GCM) before I made the switch (along with my friends) to TextSecure.

I've tried re-installing (aswell as re-exchanging keys) to no avail. What 'seems' to work (and probably placebo as its not always 100%) is to kill TextSecure (via Recents), re-open, and 50/50 messages will suddenly flow in (with old Timestamps). 

A Timestamp example will be like '30 minutes' when I know for a fact (prior to closing TextSecure) I've been staring at the Conversation window waiting for a reply. Unsure what logs/debugs things I can provide (or how) - but happy to help in anyway!

**Comment by hult-jaeger** [comment]
TS @ 2.7.1; Android @ 4.0.4

can confirm that there are no notification messages from TS itself (in the background, or whatever you want to call it). Only when TS is opened manually, it will report on new messages. This happens to me and 3 friends on mine. All of them on Android TS.

**Comment by moxie0** [comment]
@Sinlicious Thanks for the update but we'll need a debug log from before and after you kill TS to be able to take any action.

**Comment by moxie0** [comment]
@Sinlicious Also has this behavior happened (where you have to kill the app and then reopen it) within the past 4 or 5 days?

**Comment by gordon-morehouse** [comment]
>> Before secure SMS transport was removed, falling back to secure SMS appeared to ram a message  through if the data transport was not working.

> FYI, it did no such thing.

Please note my use of the word 'appeared' - in attempting to develop a workaround, I reasoned that falling back to secure SMS might get a message through when the data transport was failing, and on the few occasions I managed to experiment with it before the last release, sometimes the non-confirmed data transport messages would get confirmed pretty quickly after a secure SMS - I suspected due to the user opening TS on the remote end.

If you'll bear with me for a moment, it doesn't matter how this or that bit of code behaves to the end user, but rather how the *whole app* appears to behave. Users are accustomed to SMS being near real-time, particularly in ongoing conversations. Users will attempt to develop workarounds when TextSecure doesn't provide this. I should have clarified that comment with "... I suspect because it was causing the remote user to wake TS or the TS instance to be woken," but I didn't.

**Comment by nh2** [comment]
@moxie0 Here's a log in which I sent a message from another phone (`+*********32`) with text `2336` at `23:36` my time, and it arrived 3-4 minutes later (the receiving phone had the screen off during all this time).

https://gist.github.com/anonymous/2ae6b4135e32912c017c

Also, the sending phone did not get a delivery report until it actually arrived on the other phone.

Please let me know if this is helpful or if you need a different log.

May I ask, how do you manage to read anything out of this, given that there don't even seem to be any time stamps in this log?

**Comment by Sinlicious** [comment]
@moxie0 Thanks for the prompt reply - with regards to the first question do I simply go to Settings > Advanced > Submit Debug Log to achieve what you want? If I understand you want me to get logs specifically at:

1. After sending a few messages via Push;
2. Closing TextSecure, killing it via Recents, and re-opening.

Any other cases? I'm unsure when/how (unless I call the 'Other Parties') if/when they have sent a reply, so do you only want a log in Situation 2 after opening TextSecure resulted in delayed messages?

With regards to when this has occurred... I've only been using TextSecure for past two weeks or so, and I can't pinpoint with 100% accuracy, but I would say more recently it has had this behaviour - though I've acquired more TextSecure converts in last week - so it may have something to do with Signal users vs. Android and alike.

Let me know if you need anything else - it seems that others are also having this issue above, but also in the #970 Issue thread.

**Comment by jeremymasters** [comment]
I hadn't had this go on for awhile but it happened again last night.  Was messaging with a friend and at a seemingly random time, I sent a message, received a confirmation checkmark but the info shows a "sent" but not a "received" time.

**Comment by moxie0** [comment]
@jeremymasters If you got a delivery receipt it was delivered.

**Comment by nh2** [comment]
Is it clear what the cause / exact fix was?

**Comment by devurandom** [comment]
@nh2: Watch out for when @moxie0 pushes the v1.2.5 tag to WhisperSystems/libtextsecure-java.

**Comment by crash2kx** [comment]
thanks everyone fixing this. hope our debugs were helpful :)

**Comment by crash2kx** [comment]
Bad news. I just happened to witness this bug again. I sent my friend a message at 1pm and as soon as it arrived I received an earlier message from him from 11:42am. 

I don't know if the debug log helps but here it is anyway https://gist.github.com/anonymous/13a1cbdfec152dda2add

**Comment by nRaecheR** [comment]
Yes, same for me...2.10.2 didn't fix the bug and so it must be re-opened here...it would be better to get confirmation that the bug is actually fixed *before* closing the bug...

**Comment by rhodey** [comment]
there are existing issue reports open about messages being received late, this report was about notifications not showing up at all when receiving a message-- regardless of issues with delivery time, or delivery receipts, if you receive a message **and** no accompanying notification then this issue could possibly be reopened.

however, as noted above, GCM is not perfect, so it is possible and out of our control that sometimes a notification will not show up for received messages. this should happen very infrequently, if it happens frequently then TS should be able to improve on that.

**Comment by huegelc** [comment]
Same issue here too... Had restored TS from a titanium backup

**Comment by agrajaghh** [comment]
titanium backup is known to cause issues with TS

**Comment by gordon-morehouse** [comment]
No restore in my case - same install I've been using for months. Problems appeared after most recent update but were delayed by a day or two.

Anecdata: I managed to get one crypto-naive user to install TextSecure, he was extremely enthusiastic at first and had a couple friends install it, but uninstalled it after a couple weeks due to the delays. I know the devs know this is a serious problem, just wanted to illustrate it's a "users are bailing" level problem for people who aren't hardcore into privacy vs things working right as often as SMS does.

**Comment by moxie0** [comment]
@gordon-morehouse All the server and client bugs around this have been resolved.  At this point, if you're missing notifications it means either:

1. You've disabled background data for TextSecure or any Google service.
2. You've used titanium backup or something like it.
3. You have a bad gapps install on a rooted device.
4. You're on a network that's blocking GCM.

There's nothing we can do about any of those things, and we no longer believe there are any problems on our end.

**Comment by gordon-morehouse** [comment]
Moxie, I respect greatly what you're doing here but I don't think you're correct. I'm seeing this amongst multiple people on multiple devices and multiple cell networks, some of whom are talking to *each other* and then complaining to me about the app I told them to install!  I'm an evangelist, and as such I really want to see TextSecure work as smoothly as SMS. Right now, it simply does not for a subset of users. Maybe it is because I know a lot of poor people with older phones running crappy Android versions. I don't know.

FWIW:

1. I haven't disabled background data at all and tend to stick on 3G instead of wifi (weak wifi and textsecure play *badly* together.†
1. I have never used Titanium Backup on this device.
1. I have the stock Sprint ROM and my device is not rooted. (Seriously. Not even once, this go-around.)
1. I'm on Ting via CDMA; since the app works most of the time and so do my others, I seriously doubt GCM is being blocked. GCM being a steaming pile, now that I could buy. (Once in a while the push problems are not confined to TS - I'll have trouble with Hangouts and Hootsuite and the like, but it's much rarer than message delays with TS.)

If you don't believe there are problems, why not close the bug? I know you all have been working very hard on TS since Signal got out the door - I've seen the updates and some major improvement - but there are still problems.

I have a busy life, I don't have the debug log procedure memorized, and it is hard to coordinate log submission with other users, especially less technical folks.

I am trying to help make TS better.

† I know there's not much you can do about that but a quicker retry interval might help? TS seems to get "stuck" sometimes once one message has the "..." status, even when I drop back to 3G or walk into better range of the access point, if I have a poor wifi signal when I send a message. If that's just GCM fail, well, GCM is ... less than great.

**Comment by moxie0** [comment]
> I know there's not much you can do about that but a quicker retry interval might help? TS seems to get "stuck" sometimes once one message has the "..." status, even when I drop back to 3G or walk into better range of the access point, if I have a poor wifi signal when I send a message. If that's just GCM fail, well, GCM is ... less than great.

It sounds like you're talking about message *sending* rather than message *receiving*?  The former is completely unrelated to GCM and my understanding of the discussion in this issue.  If the problems you are receiving are related to messages not *sending*, that's a whole other issue we can look at.

> If you don't believe there are problems, why not close the bug?

FWIW, its been closed since March 31st.

**Comment by gordon-morehouse** [comment]
> It sounds like you're talking about message _sending_ rather than message _receiving_? The former is completely unrelated to GCM and my understanding of the discussion in this issue. If the problems you are receiving are related to messages not _sending_, that's a whole other issue we can look at.

Correct. I'll either initiate a conversation with a TextSecure user and there'll be a longer-than-15-min delay when I later find that their phone was in range and in use (so not sleeping), or I will be _having_ a conversation and suddenly bam. Messages get acknowledged by TS but there is no confirm. Minutes to hours later, usually, all of them arrive in a flood to the recipient. Then they become confirmed in TextSecure. I don't know the specifics of the less technical folks' problems but two complained about "losing texts" and, when prompted, their symptoms sounded similar.

I have two older Android users and an iPhone user I can try to coordinate some testing on, we're all developers.

> FWIW, its been closed since March 31st.

Touché. :)

**Comment by huegelc** [comment]
Yes Titanium Backup was the culprit. I had to unregister/register to receive notifications again.

**Comment by kb100** [comment]
I think this issue needs to be reopened. I've been experiencing this bug on Wifi for as long as I can remember, and with the update to 2.25.3 in the play store, I have now experienced it for the first time on 4G data (Verizon). The symptoms are identical to those described in this issue:

I wake up this morning and check my messages, only to get 20 notifications, some from individuals and some from group chats, all at once, timestamped from the night before.

It also happened yesterday. I switched Wifi on to watch a youtube video, then turned it off. I didn't get any notifications for 6 hours. I opened TS, still nothing, and it wasn't until I actually pressed on a specific contact that I suddenly received 6 messages (some from that contact, some from others) sporadically dated over that day. I'm not sure if this was even a foreground/background issue because I seemed to have to press on the specific contact, not just open the app. I guess it could have just taken that exact amount of time from when I opened the app to download the new messages, but it was an eerie coincidence.

Is it possible that my phone doesn't switch between wifi and 4G correctly? Or that it saves some broken wifi settings even when switching to 4G? Again I don't even know what is relevant to the problem, but the missed messages started after I switched the wifi on and off. Coincidence? I don't know.

1. Data usage settings are (and have been for years): Mobile data (checked), Limit mobile data usage (NOT checked), Alert me about data usage (checked). Under TS specifically, Restrict background data is NOT checked. Data usage shows 28.64 MB used this cycle, 0.02GB Foreground, <0.01GB background. (How to check if <0.01 means 0?)

2. I have never used Titanium backup or any other backup app or software for this device. Are there any other apps known to cause issues with TS?

3. I am running Verizon stock Galaxy S4, never rooted.

4. Does Verison's 4G network block GCM? I don't know, but I wouldn't think so.

**Comment by gordon-morehouse** [comment]
TextSecure is still extremely bad at handling switches between wifi and mobile data, but my device is ancient and I'm too tired to help debug it any further. It's still pretty broken though. $.02.

**Comment by real-or-random** [comment]
I've also encountered this several times in the past days (but can't say for sure it was some particular update) after the issue was gone for a very long time.

Note: I don't know details about the culprits with Titanium Backup. It's installed on my phone but I have never used it to restore TS nor any other app.

**Comment by kb100** [comment]
It just happened again to me (2 messages 51 minutes late, messages were sent to me right after switching to wifi), and I noticed that the messages were received as soon as I unlocked my phone. I did not have to open TS. My phone and SD card are encrypted using the built in encryption feature. Moreover, I have "lock instantly with power key" and "lock automatically after 30 minutes" turned on. *I think* the encryption key is flushed while the device is locked. 

Is it possible that TS is somehow prevented from writing to memory while the device is locked?

**Comment by McLoo** [comment]
@kb100 can you definitely say that you are not running any kind of battery save mode or battery saver app (eg greenify)

**Comment by kb100** [comment]
Power saving mode is NOT enabled and I have never installed any kind of battery saver app.

**Comment by tillschaefer** [comment]
i am also suffering from the issue, that messages are not delivered whenever the app is closed. 

i am running the latest textsecure app on cyanogenmod kitkat

- background data is activated 
- privacy seetting in cyanogen are disabled 
- i am not using a titanium backup
- i am using links2sd but have it disabled, reinstalled and reinstalled the app, still no success
 
the failure is permanent, perfectly reproducible. if the app is closed in the task manager (i.e. long pressing home), i do not get any message until i open the app. i also sometimes need to send a message myself before the message from another person in received.

**Comment by tillschaefer** [comment]
ok i have used push notification tester [1]. It seems that push is not working on my device. but why? everything else from gapps is working (play store, google sync, ...).

[1] https://play.google.com/store/apps/details?id=com.firstrowria.pushnotificationtester&hl=en

**Comment by crapp** [comment]
I have used 
> adb backup -shared -all -nosystem -apk 
> adb restore...
to backup my data before I switched from stock android to cyanogenmod 12.1 on my nexus 4 and I experience the same problems once in a while. Do I have to uninstall Textsecure completely to get this working properly?

**Comment by moxie0** [comment]
You have to reregister

On 10/07/2015 03:56 AM, Christian Rapp wrote:
> I have used
> 
>     adb backup -shared -all -nosystem -apk
>     adb restore...
>     to backup my data before I switched from stock android to
>     cyanogenmod 12.1 on my nexus 4 and I experience the same problems
>     once in a while. Do I have to uninstall Textsecure completely to get
>     this working properly?
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WhisperSystems/TextSecure/issues/2703#issuecomment-146154966>.
> 

-- 
http://www.thoughtcrime.org

**Comment by ChrisWitt** [comment]
Ran into same problem: No messages received while phone on stand by.

Turned out for me the troublemaker was a new power safe feature in Android 6.
On my Phone Samsung S5 running Cyanogen Mod 13 I had to disable battery optimization for signal to keep it active when the phone was on standy:
Settings->Apps->Settings>Battery Optimisiation>display all apps>select signal>deactivate battery optimization for signal

See this article for more information, it final gave me the needed clue: http://arstechnica.com/gadgets/2015/10/android-6-0-marshmallow-thoroughly-reviewed/9/

I know this thread was created for android 4 but it was the first describing the same problem when searching for a solution.
